uvicorn/_types.py

```diff
@@ -1,14 +1,44 @@
+import asyncio
 import types
-import typing
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Iterable,
+    MutableMapping,
+    Optional,
+    Protocol,
+    Tuple,
+    Type,
+    Union,
+)
+
+if TYPE_CHECKING:
+    from uvicorn.config import Config
+    from uvicorn.server_state import ServerState
 
 # WSGI
-Environ = typing.MutableMapping[str, typing.Any]
-ExcInfo = typing.Tuple[
-    typing.Type[BaseException], BaseException, typing.Optional[types.TracebackType]
-]
-StartResponse = typing.Callable[
-    [str, typing.Iterable[typing.Tuple[str, str]], typing.Optional[ExcInfo]], None
-]
-WSGIApp = typing.Callable[
-    [Environ, StartResponse], typing.Union[typing.Iterable[bytes], BaseException]
-]
+Environ = MutableMapping[str, Any]
+ExcInfo = Tuple[Type[BaseException], BaseException, Optional[types.TracebackType]]
+StartResponse = Callable[[str, Iterable[Tuple[str, str]], Optional[ExcInfo]], None]
+WSGIApp = Callable[[Environ, StartResponse], Union[Iterable[bytes], BaseException]]
+
+
+class WebProtocol(Protocol):
```

Comment:
Here, I'm not sure if we should actually call it `WebProtocolProtocol`, because the `Protocol` is from typing. :thinking: